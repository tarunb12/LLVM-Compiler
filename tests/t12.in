// Recursion test (gcd)

int gcd(int m, int n) {
    if (n != 0) return gcd(n, m % n)
    else return m
}

int main() {
    string gcd_spec = "gcd(%d, %d) = %d\n"

    printf(gcd_spec, -30, 45, gcd(-30, 45))
    printf(gcd_spec, 8, 23, gcd(8, 23))
    printf(gcd_spec, 36, 404, gcd(36, 404))
    printf(gcd_spec, 83, 93, gcd(83, 93))
    printf(gcd_spec, 203, -1102, gcd(203, -1102))
    printf(gcd_spec, 163231, 135749, gcd(163231, 135749))
    
    return 0
}