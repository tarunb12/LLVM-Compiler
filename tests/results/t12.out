Error: File "tests/t12.in": Invalid return type specified in the function fibonacci: "unit", expected "int" in "return c".